cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -737,22 +742,50 @@ public QueryHandle createNamespace(Id.Namespace namespace) throws ExploreExcepti
       SessionHandle sessionHandle = null;
       OperationHandle operationHandle = null;
 
+      // Get custom Hive database name if one was  provided
+      customHiveDatabase = getCustomHiveDatabaseIfExists(namespace.getId());
       try {
         sessionHandle = cliService.openSession("", "", sessionConf);
-
-        String database = getHiveDatabase(namespace.getId());
-        // "IF NOT EXISTS" so that this operation is idempotent.
-        String statement = String.format("CREATE DATABASE IF NOT EXISTS %s", database);
-        operationHandle = doExecute(sessionHandle, statement);
-        QueryHandle handle = saveReadOnlyOperation(operationHandle, sessionHandle, sessionConf, statement, database);
-        LOG.info("Creating database {} with handle {}", namespace, handle);
+        QueryHandle handle;
+        if (Strings.isNullOrEmpty(customHiveDatabase)) {
```

Comment:
To me, setting the hive db is orthogonal to whether we should auto-create and delete the database. Should they be 2 separate settings?